MapboxNavigation/ExitView.swift

```diff
@@ -53,13 +53,20 @@ class ExitView: StylableView {
             invalidateIntrinsicContentSize()
         }
     }
+    
     var pointSize: CGFloat {
         didSet {
             exitNumberLabel.font = exitNumberLabel.font.withSize(pointSize * ExitView.labelFontSizeScaleFactor)
             rebuildConstraints()
         }
     }
     
+    override var hashValue: Int {
+        get {
+             return [foregroundColor?.cgColor.hashValue ?? 0, backgroundColor?.cgColor.hashValue ?? 0, exitText?.hashValue ?? 0, pointSize.hashValue, side.hashValue].reduce(0, +)
```

Comment:
Three things:

1. Why are you keying off of `cgColor`? `UIColor's` hash conformance has good consistency across instances. E.g: Two `UIColor` objects with the same RGB values will result in the same hash.
2. Using addition here is a problem. The reason why XOR is considered the best solution (out of all common boolean/arithmetic algebra functions) for combining hashes is because it has [better distribution](http://eigenjoy.com/2011/05/04/why-is-xor-the-default-way-to-combine-hashes) then other common boolean operations, and it's considered preferable to addition because you're guaranteed to [stay in the same bit-space](https://en.wikipedia.org/wiki/Integer_overflow), as well as having better [entropy properties](https://stackoverflow.com/questions/5533971/does-adding-two-hash-values-generate-a-valid-hash#comment9328749_5533985). Some folks mitigate this by doing some remainder math when they reach an overflow scenario, but I'd be worried about that introducing modulo-bias.
3. I didn't notice that some of those properties are optionals that does change things, but I would still argue that a more functional representation here is both doable and would read nicer than what we have currently.

``` swift
return [foregroundColor, backgroundColor, exitText, pointSize, side].compactMap({ $0?.hashValue }).reduce(0, ^)

// Or, if you're worried about performance (although I don't think there would be any real impact, and it wouldn't be quite as readable) 

return [foregroundColor, backgroundColor, exitText, pointSize, side].reduce(0, { $0 ^ $1?.hashValue ?? 0})
```

Reply:
Supplemental: 

If, for some reason, we _do_ decide to keep arithmetic combination of hashes, we should _absolutely_ use [Overflow Operators](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/AdvancedOperators.html#//apple_ref/doc/uid/TP40014097-CH27-ID37) (`&+`) over a regular addition operator, as the regular one will produce a swift error if it encounters an overflow situation.

Reply:
The problem is that we're mixing types here and that calling compactMap on an array of mixed types produces some compiler errors.